pkg/kbld/config/config.go

```diff
@@ -322,7 +329,8 @@ func UniqueImageOverrides(overrides []ImageOverride) []ImageOverride {
 	for _, override := range overrides {
 		var found bool
 		for _, addedOverride := range result {
-			if addedOverride == override {
+			// using DeepEqual instead of '==' since ImageOverride contains a slice
+			if reflect.DeepEqual(addedOverride, override) {
```

Comment:
Yeah, we want to look into what the possible "conflict" use cases there are... and how we want each to be resolved.  And if need be, make the appropriate changes.

Reply:
I think it looks like this:

- no dups ==> no overrides removed
- matching overrides (exact duplicates) ==> duplicates removed
- equal "image[+imagerepo]": 
  - preresolved > !preresolved — respect presence of "preresolved" by not re-resolving
  - !tagselection > tagselection — presume "newImage" without tagselection is more precise and therefore wins
  - latest(tagselection) — for `semver` (current only strategy supported), whichever selects the largest/latest version value.
- first override specified wins — if a set of overrides are not unique by this point, then take the first.

Reply:
oh wait... I think I misunderstood our intent, here.

- on reflection, I see that: _The order_ of overrides is the only thing that determines which one applies (i.e. first _always_ wins, regardless) — this is **not** changing
- ... rather ...: There are a couple of cases where additional meta are added (notably "preresolve"s). The concern here is making sure that we don't lose any meta on the way through de-duping. 
- also noted while reading the surrounding code more carefully: we don't actually call `UniqueImageOverrides()` when emitting an `imgpkg`-flavored lock file. I believe this is an oversight.

So, I'll review this flow with my revised understanding, today.

Reply:
Back to the original question:
> do we have any funky behaviour when we have same image multime times with different metas?

The answer is no... mostly because in all the code paths that lead to `UniqueImageOverrides()`, _right now_, none of the overrides have meta set.

This led me to step back and reflect on the overall workflow...

With this PR, we are propagating meta through `imgpkg` lock files.
However, we are _not_ doing so with plain `kbld` lock files.
Technically, this was not requested in either the issue or intended in this PR.

That said, if it's not unreasonable that users may which to use `kbld` lock files during development and then switch to `imgpkg` lock files when bundling up the config+images, then it would make sense to ensure that meta are carried through the `kbld` lock file flow.

I'm capturing this additional scope in a separate issue and it can be prioritized.